test/acceptor_SUITE.erl

```diff
@@ -159,6 +160,102 @@ misc_info(_) ->
+	{ok, _} = gen_tcp:connect("localhost", Port1, [binary, {active, false}, {packet, raw}]),
+	receive after 250 -> ok end,
+	ranch:set_protocol_options({misc_info_embedded, tcp}, [{remove, false, 2500}]),
+	receive after 250 -> ok end,
+	{ok, _} = gen_tcp:connect("localhost", Port1, [binary, {active, false}, {packet, raw}]),
+	{ok, _} = gen_tcp:connect("localhost", Port1, [binary, {active, false}, {packet, raw}]),
+	receive after 250 -> ok end,
+	%% Confirm the info returned by Ranch is correct.
+	[
+		{{misc_info_embedded, act}, [
+			{pid, Pid2},
+			{ip, _},
+			{port, Port2},
+			{num_acceptors, 2},
+			{max_connections, infinity}, %% Option was modified.
+			{active_connections, 0},
+			{all_connections, 0},
+			{transport, ranch_tcp},
+			{transport_options, [{num_acceptors, 2}]},
+			{protocol, active_echo_protocol},
+			{protocol_options, {}}
+		]},
+		{{misc_info_embedded, ssl}, [
+			{pid, Pid3},
+			{ip, _},
+			{port, Port3},
+			{num_acceptors, 3},
+			{max_connections, 1024},
+			{active_connections, 0},
+			{all_connections, 0},
+			{transport, ranch_ssl},
+			{transport_options, [{num_acceptors, 3}|Opts]},
+			{protocol, echo_protocol},
+			{protocol_options, [{}]}
+		]},
+		{{misc_info_embedded, tcp}, [
+			{pid, Pid1},
+			{ip, _},
+			{port, Port1},
+			{num_acceptors, 1},
+			{max_connections, 1024},
+			{active_connections, 2},
+			{all_connections, 5},
+			{transport, ranch_tcp},
+			{transport_options, [{num_acceptors, 1}]},
+			{protocol, remove_conn_and_wait_protocol},
+			{protocol_options, [{remove, false, 2500}]} %% Option was modified.
+		]}
+	] = ranch_listenergroup_info(misc_info_embedded),
+	%% Get acceptors.
+	[_] = ranch:procs({misc_info_embedded, tcp}, acceptors),
+	[_, _] = ranch:procs({misc_info_embedded, act}, acceptors),
+ 	[_, _, _] = ranch:procs({misc_info_embedded, ssl}, acceptors),
+	%% Get connections.
+	[_, _, _, _, _] = ranch:procs({misc_info_embedded, tcp}, connections),
+	[] = ranch:procs({misc_info_embedded, act}, connections),
+ 	[] = ranch:procs({misc_info_embedded, ssl}, connections),
+	%% Stop embedded tcp listener and ensure it is gone.
+	ok = embedded_sup:stop_listener(SupPid, {misc_info_embedded, tcp}),
+	timer:sleep(500),
+	[{{misc_info_embedded, act}, _}, {{misc_info_embedded, ssl}, _}]=
+		ranch_listenergroup_info(misc_info_embedded),
+	%% Stop embedded act listener and ensure it is gone.
+	ok = embedded_sup:stop_listener(SupPid, {misc_info_embedded, act}),
+	timer:sleep(500),
+	[{{misc_info_embedded, ssl}, _}]=
+		ranch_listenergroup_info(misc_info_embedded),
+	%% Stop embedded ssl listener and ensure it is gone.
+	ok = embedded_sup:stop_listener(SupPid, {misc_info_embedded, ssl}),
+	timer:sleep(500),
+	[]=ranch_listenergroup_info(misc_info_embedded),
+	%% Stop embedded supervisor.
+	embedded_sup:stop(SupPid),
+	ok.
+
+ranch_listenergroup_info(ListenerGroup) ->
+	lists:sort([L || L={{G, _}, _} <- ranch:info(), G=:=ListenerGroup]).
```

Comment:
Prefix this with `do_` please, if I ever switch to the same system as in Cowboy this'll avoid considering this function to be a test.